lib/loggers/index.js

```diff
@@ -51,31 +52,42 @@ const container = new winston.Container();
 // override get to use a wrapper so loggers that are retrieved via `.get()`
 // prior to logger initialization will receive the updated configuration
 // when logging after initialization
-container._get = container.get;
+// container._get = container.get;
 container._add = container.add;
-container.get = container.add = function(id) {
-  const existing = container.loggers[id];
-  let logger = container._get.apply(container, arguments);
+container.add = function(id) {
+  const existing = container.has(id);
+  let logger = container._add.apply(container, arguments);
   if(!existing) {
     const wrapper = Object.create(logger);
-    wrapper.log = function(level, msg, meta) {
-      // merge per-logger and per-log meta and call parent logger
-      meta = Object.assign({}, this.meta, meta);
-      return Object.getPrototypeOf(wrapper).log.apply(
-        wrapper, [level, msg, meta]);
+    wrapper.log = function(level, msg, ...meta) {
+      // get default meta to ensure it is always applied
+      const {defaultMeta} = this;
+      // use `Object.getPrototypeOf` because the prototype may change
+      // while the logging subsystem initializes
+      const log = Object.getPrototypeOf(wrapper).log;
+      if(arguments.length === 1) {
+        return log.apply(wrapper, [{...defaultMeta, ...level}]);
+      }
+      if(meta.length === 0) {
+        meta.push({defaultMeta});
+      } else {
+        meta[0] = {...defaultMeta, ...meta[0]};
```

Comment:
I think it is safe to just to this on L74 even if `meta.length === 0`, and thus avoid the special casing.  See:

https://repl.it/@mattcollier/OfficialFlippantTechnologies#index.js

Reply:
Agreed.